### PR TITLE
[RHCLOUD-13364] refactor: make the spec match the back end's endpoints

### PR DESCRIPTION
**Pull request by MikelAlejoBR**
The things that have been done:

- Fix license URL to use the "https" scheme.
- Add the ability to authorize using headers.
- Include all the environments: local, staging and production.
- Improve some of the descriptions for the endpoints.
- "400" returns for many endpoints that lacked them.
- Update the models for the payloads and responses, which weren't 100%
  accurate.
- Add a missing "/application_authentications/{id}/authentications" endpoint.

## Links

[[RHCLOUD-9245]](https://issues.redhat.com/browse/RHCLOUD-9245)
[[RHCLOUD-18477]](https://issues.redhat.com/browse/RHCLOUD-18477)
[[RHCLOUD-13364]](https://issues.redhat.com/browse/RHCLOUD-13364)

### CODE REVIEW

**Comment by MikelAlejoBR** [comment]
The SonarQube failure is related to this PR: https://github.com/RedHatInsights/sources-api-go/pull/335 .

**Comment by lpichler** [comment]
I am not asking to split up it - but there are too much changes in this PR. 

There is good article - https://smallbusinessprogramming.com/optimal-pull-request-size/

**Review by lpichler — APPROVED** [review]

File: public/openapi-3-v3.1.json [hunk 1]
```diff
@@ -139,90 +154,82 @@
...
-        "operationId": "deleteApplicationAuthentication",
-        "description": "Deletes a ApplicationAuthentication object",
+      }
+    },
+    "/applications_authentications/{id}/authentications": {
+      "get": {
+        "summary": "List authentications for the given application authentication",
+        "operationId": "listApplicationAuthenticationsAuthentications",
+        "description": "Returns an array of authentications",
         "parameters": [
```
Comment (petracihalova) [inline public/openapi-3-v3.1.json:203]: I think in parameters is missing something like 
```
{
"$ref": "#/components/parameters/ID"
}
```

File: public/openapi-3-v3.1.json [hunk 2]
```diff
@@ -489,6 +493,9 @@
               }
             }
           },
+          "400": {
+            "$ref": "#/components/responses/BadRequest"
+          },
           "404": {
```
Comment (petracihalova) [inline public/openapi-3-v3.1.json:502]: here can be the one line version to have it same as is on other places
`"$ref": "#/components/responses/NotFound"`

I noticed that the "more lines version" is several times in document ... I think it is not wrong but maybe we can have it in same way for all occurrences

File: public/openapi-3-v3.1.json [hunk 3]
```diff
@@ -1487,9 +1372,6 @@
           "400": {
             "$ref": "#/components/responses/BadRequest"
           },
-          "401": {
-            "$ref": "#/components/responses/Unauthorized"
-          },
           "404": {
             "$ref": "#/components/responses/NotFound"
           }
```
Comment (petracihalova) [inline public/openapi-3-v3.1.json:1372]: in the end of `/sources/{id}/rhc_connections` and` /rhc_connections/{id}/sources` we have tags with values "sources" and "rhc-connections" ... it causes that both methods are displayed in endpoints related with both of this groups (example on picture) ... but we follow this approach only here and not for others endpoints ... I think, it should be same in whole spec

image

**Review by petracihalova — CHANGES_REQUESTED** [review]
Nice job Mikel, this is really lot of changes. I have only tiny comments.

File: public/openapi-3-v3.1.json (hunk 1 above)
Comment (MikelAlejoBR) [reply public/openapi-3-v3.1.json:203]: Done, thanks!

File: public/openapi-3-v3.1.json (hunk 2 above)
Comment (MikelAlejoBR) [reply public/openapi-3-v3.1.json:502]: Right, I must have missed those. Fixed!

File: public/openapi-3-v3.1.json (hunk 3 above)
Comment (MikelAlejoBR) [reply public/openapi-3-v3.1.json:1372]: All right, I've removed the redundant tags!

**Review by petracihalova — APPROVED** [review]
Awesome! Let's merge it 😁

**Pull request merged**